tests/vfolders/path_matcher.py

```diff
@@ -0,0 +1,194 @@
+        == ["language0", "language1"])
+
+
+@pytest.mark.pootle_vfolders
+@pytest.mark.django_db
+def test_vfolder_path_matcher_projects(project0, project1):
+    vfolder = VirtualFolder.objects.create(
+        name="avfolder",
+        filter_rules="FOO,BAR")
+    vfolder.projects.add(project0)
+    vfolder.projects.add(project1)
+    # no languages set so no stores
+    assert list(vfolder.path_matcher.existing_stores) == []
+    assert (
+        sorted(vfolder.path_matcher.projects.values_list("code", flat=True))
+        == ["project0", "project1"])
+    vfolder.all_projects = True
+    vfolder.save()
+    assert vfolder.path_matcher.projects is None
+    assert list(vfolder.path_matcher.languages) == []
+    # no languages set so no stores
+    assert list(vfolder.path_matcher.existing_stores) == []
+    vfolder.all_projects = False
+    vfolder.save()
+    # didnt forget lang preferences
+    vfolder.all_projects = False
+    vfolder.save()
+    assert (
+        sorted(vfolder.path_matcher.projects.values_list("code", flat=True))
+        == ["project0", "project1"])
+
+
+@pytest.mark.pootle_vfolders
+@pytest.mark.django_db
+def test_vfolder_path_matcher_all_proj_lang():
+    vfolder = VirtualFolder.objects.create(
+        name="avfolder",
+        all_projects=True,
+        all_languages=True,
+        filter_rules="*")
+    return
+    assert (
+        list(vfolder.path_matcher.existing_stores.order_by("pk"))
+        == list(Store.objects.order_by("pk")))
+    vfolder.all_languages = False
+    vfolder.save()
+    assert (
+        list(vfolder.path_matcher.existing_stores.order_by("pk"))
+        == [])
+    vfolder.all_languages = True
+    vfolder.all_languages = False
+    vfolder.save()
+    assert (
+        list(vfolder.path_matcher.existing_stores.order_by("pk"))
+        == [])
+
+
+@pytest.mark.pootle_vfolders
+@pytest.mark.django_db
+def test_vfolder_path_matcher_some_proj_lang(tp0, language1):
+    vfolder = VirtualFolder.objects.create(
+        name="avfolder",
+        filter_rules="*")
+    vfolder.languages.add(tp0.language)
+    vfolder.projects.add(tp0.project)
+    # m2m handler?
+    vfolder.save()
+    assert (
+        list(vfolder.path_matcher.existing_stores.order_by("pk"))
+        == list(tp0.stores.order_by("pk")))
+    vfolder.languages.add(language1)
+    vfolder.save()
+    assert (
+        list(vfolder.path_matcher.existing_stores.order_by("pk"))
+        == list(
+            Store.objects.filter(
+                translation_project__project=tp0.project,
+                translation_project__language__in=[
+                    tp0.language,
+                    language1]).order_by("pk")))
+
+VF_RULE_TESTS = [
+    "*",
+    "store.po",
+    "*/subdir0/*"]
+
+
+@pytest.fixture(params=VF_RULE_TESTS)
+def vf_rules(request):
+    return request.param
+
+
+@pytest.mark.pootle_vfolders
+@pytest.mark.django_db
+def test_vfolder_path_matcher_get_rule_regex(vfolder0, vf_rules):
```

Comment:
as a rule parametrized fixturs are always plural - we might want to think about namespacing them in some wayin the future